### PR TITLE
Convert `driver` crate into a `bin`

### PR DESCRIPTION
**Pull request by nlordell**
This PR turns `driver` crate into a `bin` crate instead of a `lib` crate. It also corrects all the clippy warnings that surfaced because of it.

Documention, CI and Dockerfiles were also updated.

### Test Plan

CI + :pray:

### CODE REVIEW

**Comment by e00E** [comment]
For the batched auction reader I am writing a test in the `e2e` folder/crate that uses types from driver. If this is a bin does that mean I can no longer do that? Or maybe it means I should move that test into the driver crate (but it still needs ganache to run).

File: driver/src/models/account_state.rs [hunk 1]
```diff
@@ -68,19 +30,60 @@ impl AccountState {
...
+                state_index: U256::zero(),
+                state_hash: H256::zero(),
+                balances: HashMap::new(),
+                num_tokens: std::u16::MAX,
+            };
+            for order in orders {
+                state.increment_balance(order.sell_token, order.account_id, order.sell_amount);
+            }
+            state
+        }
```
Comment (bh2smith) [inline driver/src/models/account_state.rs]: I like this! This is nice.

**Comment by nlordell** [comment]
> For the batched auction reader I am writing a test in the `e2e` folder/crate that uses types from driver. If this is a bin does that mean I can no longer do that? Or maybe it means I should move that test into the driver crate (but it still needs ganache to run).

Is there value in making this a e2e test instead of a unit test with mocked contract responses?

I think a valid e2e test would be to set the `AUCTION_DATA_BATCH_SIZE` to a low number and then run an auction with a large number of orders (relative to `AUCTION_DATA_BATCH_SIZE`) and then assert that a solution was found.

**Comment by e00E** [comment]
I think you are right. Don't let that stop this PR.

**Pull request merged**